docs/development/pigs/pig-030.rst

```diff
@@ -0,0 +1,447 @@
+
+    model = crab_point_source_model()
+
+    model_npred = NPredSystematicsModel.from_dataset(
+        model=model,
+        dataset=dataset,
+        edisp_modifier=edisp_modifier,
+    )
+    dataset.models = model_npred
+
+4. All of the above at once
+
+.. code-block:: python
+
+
+    model_npred = NPredSystematicsModel.from_dataset(
+        model=model,
+        dataset=dataset,
+        exposure_modifier=exposure_modifier,
+        psf_modifier=psf_modifier,
+        edisp_modifier=edisp_modifier,
+    )
+    dataset.models = model_npred
+
+5. PSF modification with increasing Energy
+
+.. code-block:: python
+
+    # apply scaling linear as scaling = slope * energy + intersect
+    psf_parameter_slope = Parameter('psf_scaling_slope', 1)
+    psf_parameter_slope.prior = GaussianPrior(mu=1, sigma=0.2)
+    psf_parameter_intersect = Parameter('psf_scaling_intersect', 1)
+    psf_parameter_intersect.prior = GaussianPrior(mu=1, sigma=0.2)
+
+    offset_effect = LinearScalingEffect(depends_on_axes=['energy_true']) # change scaling with energy_true
+
+    psf_modifier = Modifier([psf_parameter_slope, psf_parameter_intersect], offset_effect, axis='rad') # applied to rad axis
+
+    model = crab_point_source_model()
+
+    model_npred = NPredSystematicsModel.from_dataset(
+        model=model,
+        dataset=dataset,
+        psf_modifier=psf_modifier,
+    )
+    dataset.models = model_npred
+
+
+Implementation
+==============
+
+Modifier
+--------
+
+A ``Modifier`` links a single instance or list of ``Parameter`` instances to an ``Effect``.
+The modifier is responsible for calling the effect with the current parameter values and applying it to
+either the IRF *data* or its *axis* coordinates (e.g. shifting the ``energy`` axis).
+In the latter case the transformation defined by the ``Effect`` is applied by interpolating the ``IRF`` to new transformed coordinates.
+
+.. code-block:: python
+
+    class Modifier:
+        def __init__(self, parameters, effect, axis=None):
+            self.parameters = parameters
+            self.effect = effect
+            self.axis = axis
+
+        def __call__(self, irf):
+            if self.axis is None:
+                # Direct modification of data
+                return self.effect(irf, self.parameters)
+
+            # Axis-based coordinate transformation
+            coords = self.effect(irf.geom, self.parameters, self.axis)
+            coords_scaled = MapCoord.create(coords, frame=irf.geom.frame)
+            new_data = irf.interp_by_coord(coords_scaled)
+            return irf.copy(data=new_data)
+
+Effects
+-------
+
+``Effects`` define the mathematical transformation applied to the ``IRF`` *data* or *coordiantes*/*axes*.
+Moreover, ``Effects`` can depend on one or multiple axes, which leads to parameterized transformations, e.g. ``f(enery, offset, ...)``.
+We distinguish between ``ScalingEffects`` and ``ShiftingEffects``.
```

Comment:
Note that for spatial uncertainty there could be a rotation effect as well e.g. to account for astrometry errors